File: lang/php/src/fun/vac/util/SequenceBuilder.php
<?php
    namespace fun\vac\util\SequenceBuilder;

    require_once "fun/vac/util/RandomFactory.php";
    require_once "fun/vac/util/Sequences.php";

    use fun\vac\util\RandomFactory;
    use fun\vac\util\Sequences;

    function packIncreasing(array &$arr): void {
        RandomFactory\seed();
        $arr[0] = RandomFactory\integerN(1, 3);
        for ($i = 1, $length = count($arr); $i < $length; $i++) {
            $arr[$i] = $arr[$i - 1] + RandomFactory\integerN(1, 3);
        }
    }

    function packRandom(array &$arr): void {
        RandomFactory\seed();
        foreach ($arr as &$v) {
            $v = RandomFactory\generateInteger();
        }
    }

    function packDecreasing(array &$arr): void {
        packIncreasing($arr);
        Sequences.reverse($arr);
    }
?>

File: lang/ruby/src/fun/vac/util/SequenceBuilder.rb
require "fun/vac/util/RandomFactory"
require "fun/vac/util/Sequences"

module SequenceBuilder

    def self.packIncreasing(arr)
        RandomFactory.seed()
        arr[0] = RandomFactory.integerN(1, 3)
        for i in 1 ... arr.length
            arr[i] = arr[i - 1] + RandomFactory.integerN(1, 3)
        end
    end

    def self.packRandom(arr)
        RandomFactory.seed()
        arr.each_index do |i|
            arr[i] = RandomFactory.generateInteger()
        end
    end

    def self.packDecreasing(arr)
        packIncreasing(arr)
        Sequences.reverse(arr)
    end
end
